util/cache/redis.go

```diff
@@ -26,6 +30,42 @@ type redisCache struct {
 	expiration time.Duration
 	client     *redis.Client
 	cache      *rediscache.Cache
+	compress   bool
+}
+
+func (r *redisCache) marshal(obj interface{}) ([]byte, error) {
+	buf := bytes.NewBuffer([]byte{})
+	var w io.Writer = buf
+	if r.compress {
+		w = gzip.NewWriter(buf)
+	}
+	encoder := json.NewEncoder(w)
+
+	if err := encoder.Encode(obj); err != nil {
+		return nil, err
+	}
+	if flusher, ok := w.(interface{ Flush() error }); ok {
+		if err := flusher.Flush(); err != nil {
+			return nil, err
+		}
+	}
+	return buf.Bytes(), nil
+}
+
+func (r *redisCache) unmarshal(data []byte, obj interface{}) error {
+	buf := bytes.NewReader(data)
+	var reader io.Reader = buf
+	if r.compress {
+		if r, err := gzip.NewReader(buf); err != nil {
+			if err == gzip.ErrHeader {
+				return ErrCacheMiss
+			}
+			return err
+		} else {
+			reader = r
+		}
+	}
+	return json.NewDecoder(reader).Decode(obj)
```

Comment:
What will happen if someone switches _off_ compression? Seems like there will be errors initially and then (presumably) the cache will be re-populated with plaintext content.

Would including the config flag in the cache keys provide a better user experience?

Reply:
Agree , I totally forgot about switches off compression . Changed approach to use different key when compression is enabled.